src/language-html/print/angular-icu-expression.js

```diff
@@ -0,0 +1,62 @@
+import {
+  softline,
+  line,
+  indent,
+  join,
+  group,
+} from "../../document/builders.js";
+import { getLeadingAndTrailingHtmlWhitespace } from "../utils/index.js";
+import { printClosingTagEnd, printOpeningTagStart } from "./tag.js";
+
+/*
+  <span i18n>
+    Updated {minutes, plural,
+      =0 {just now}
+      =1 {one minute ago}
+      other {{{minutes}} minutes ago}
+    }
+  </span>
+*/
+
+function printAngularIcuExpression(path, options, print) {
+  const { node } = path;
+  return [
+    printOpeningTagStart(node, options),
+    group([
+      node.switchValue.trim(),
+      ", ",
+      node.clause,
+      node.cases.length > 0
+        ? [",", indent([line, join(line, path.map(print, "cases"))])]
+        : "",
+      softline,
+    ]),
+    printClosingTagEnd(node, options),
+  ];
+}
+
+function printAngularIcuCase(path, options, print) {
+  const { node } = path;
+  return [
+    node.value,
+    " {",
+    group([
+      indent([
+        softline,
+        path.map(({ node }) => {
+          // FIXME: We need a fundamental solution.
+          // Like this: https://github.com/prettier/prettier/blob/c1ebae2e8303841e59bc02e9e719a93e964ab0e9/src/language-html/print-preprocess.js#L254
+          if (node.type === "text") {
+            const { text } = getLeadingAndTrailingHtmlWhitespace(node.value);
```

Comment:
Isn't this just ` htmlWhitespaceUtils.trim()`?

Reply:
Will this work?

```js
path.map(({ node }) => node.type === "text" && htmlWhitespaceUtils.trim(node.value) ? print() : "", "expression")
```

Reply:
No

```diff
<span i18n
  >Updated:
  {minutes, plural,
    =0 {just now}
    =1 {one minute ago}
-   other {
-     {{minutes}} minutes ago by {gender, select,
-       male {male}
-       female {female}
-       other {other}
-     }
-   }
+   other {{{minutes}} minutes ago by }
        }
      </span>
```

Reply:
But `getLeadingAndTrailingHtmlWhitespace` is exactly the same as `htmlWhitespaceUtils.trim`

Reply:
Ah, my code is wrong, should be

```js
path.map(({ node }) => node.type === "text" && !htmlWhitespaceUtils.trim(node.value) ? "" : print(), "expression")
```